vimscript.html.markdown

```diff
@@ -0,0 +1,572 @@
+echo  v:true && v:false       |" Logical AND
+echo  v:true || v:false       |" Logical OR
+echo  ! v:true                |" Logical NOT
+echo  v:true ? 'yes' : 'no'   |" Ternary operator
+```
+
+
+### Strings
+See `:h String`
+
+An ordered zero-indexed sequence of bytes. The encoding of text into bytes
+depends on the option `:h 'encoding'`.
+
+```vim
+" Literal constructors
+echo  "Hello world\n"   |" The last two characters stand for newline
+echo  'Hello world\n'   |" The last two characters are literal
+echo  'Let''s go!'      |" Two single quotes become one quote character
+```
+
+Single-quote strings take all characters are literal, except two single quotes,
+which are taken to be a single quote in the string itself. See `:h expr-quote`
+for all possible escape sequences.
+
+```
+" String concatenation
+echo  'Hello ' .  'world'  |" String concatenation
+echo  'Hello ' .. 'world'  |" String concatenation
+
+" String indexing
+echo  'Hello'[0]           |" First byte
+echo  'Hello'[1]           |" Second byte
+echo  'Hello'[-1]          |" Last byte
+echo  'Hello'[-2]          |" Second to last byte
+echo  'Hellö'[4]           |" Returns a byte, not the character 'ö'
+
+" Substrings
+echo  'Hello'[:]           |" Copy of entire string
+echo  'Hello'[1:3]         |" Substring, second to fourth byte
+echo  'Hello'[1:-2]        |" Substring until second to last byte
+echo  'Hello'[1:]          |" Substring with starting index
+echo  'Hello'[:2]          |" Substring with ending index
+```
+
+A negative index is relative to the end of the string. See `:h
+string-functions` for all string-related functions.
+
+### Lists
+See `:h List`
+
+An ordered zero-indexed heterogeneous sequence of arbitrary Vim script objects.
+
+```vim
+" Literal constructor
+echo  []                   |" Empty list
+echo  [1, 2, 'Hello']      |" List with elements
+echo  [[1, 2], 'Hello']    |" Lists can be nested arbitrarily
+
+" List concatenation
+echo  [1, 2] + [3, 4]      |" Creates a new list
+
+" List indexing, negative is relative to end of list (:h list-index)
+echo  [1, 2, 3, 4][2]      |" Third element
+echo  [1, 2, 3, 4][-1]     |" Last element
+
+" List slicing (:h sublist)
+echo  [1, 2, 3, 4][:]      |" Shallow copy of entire list
+echo  [1, 2, 3, 4][:2]     |" Sublist until third item (inclusive)
+echo  [1, 2, 3, 4][2:]     |" Sublist from third item (inclusive)
+echo  [1, 2, 3, 4][:-2]    |" Sublist until second-to-last item (inclusive)
+```
+
+All slicing operations create new lists. To modify a list in-place use list
+functions (`:h list-functions`) or assign directly to an item (see below about
+variables).
+
+### Dictionaries
+See `:h Dictionary`
+
+An unordered sequence of key-value pairs, keys are always strings (numbers are
+implicitly converted to strings).
+
+```vim
+" Dictionary literal
+echo  {}                       |" Empty dictionary
+echo  {'a': 1, 'b': 2}         |" Dictionary literal
+echo  {'x': {'a': 1, 'b': 2}}  |" Nested dictionary
```

Comment:
I'd add that you can make multiline dictionaries by prefixing new lines with a backslash:

```vim
let foo = {
  \ 'key': 'value',
  \ 'another_key': 'another_value'
\}
```

This is not immediately obvious but is very helpful for code readability as well as used frequently in plugin configuration.

Reply:
Good point. It should then also be mentioned for multi-line lists.

Reply:
This should go together with command concatenation `|`, line continuations `\` are a generic syntactic element, not limited to lists, dicts, etc, you can use them for commands, `if` statements etc.